plugins/dac_data_manager.c

```diff
@@ -479,44 +477,28 @@ static int analyse_wavefile(struct dac_data_manager *manager,
 			}
 			replicate_tx_data_channels(tx_data, tx_channels);
 
-			switch (tx_channels) {
-			case 1:
-				for (i = 0 ; i < size; i++) {
-					sample_16[i] = convert(scale, tx_data[0].re[i], offset);
-				}
-				break;
-			case 2:
-				for (i = 0 ; i < size; i++) {
-					sample_32[i] = ((unsigned int) convert(scale, tx_data[0].im[i], offset) << 16) |
-						       ((unsigned int) convert(scale, tx_data[0].re[i], offset) << 0);
-				}
-				break;
-			case 4:
-				for (i = 0 ; i < size; i++) {
-					sample[i] = ((unsigned long long) convert(scale, tx_data[1].im[i], offset) << 48) |
-						    ((unsigned long long) convert(scale, tx_data[1].re[i], offset) << 32) |
-							((unsigned long long) convert(scale, tx_data[0].im[i], offset) << 16) |
-							((unsigned long long) convert(scale, tx_data[0].re[i], offset) << 0);
-				 }
-				break;
-			case 8:
-				for (i = 0, j = 0; i < size; i++) {
-					sample[j++] = ((unsigned long long) convert(scale, tx_data[3].im[i], offset) << 48) |
-						    ((unsigned long long) convert(scale, tx_data[3].re[i], offset) << 32) |
-							((unsigned long long) convert(scale, tx_data[2].im[i], offset) << 16) |
-							((unsigned long long) convert(scale, tx_data[2].re[i], offset) << 0);
-					sample[j++] = ((unsigned long long) convert(scale, tx_data[1].im[i], offset) << 48) |
-						    ((unsigned long long) convert(scale, tx_data[1].re[i], offset) << 32) |
-							((unsigned long long) convert(scale, tx_data[0].im[i], offset) << 16) |
-							((unsigned long long) convert(scale, tx_data[0].re[i], offset) << 0);
+			int ch = 0;
+			int sample_i = 0;
+			int tx_data_end = (tx_channels % 2 == 0) ? (tx_channels/2) : tx_channels;
```

Comment:
Just a small thing: you could consider variables `ch`, `sample_i `and `tx_data_end `as unsigned. These variables won't be holding any negative values as it doesn't make sense. Therefore, I would make the unsigned or size_t (which is pretty much the same thing).